### DISCUSSION TITLE
Is it possible to hide a plot ?

### DISCUSSION THREAD

**Discussion by forna** [highfestiva/finplot#340] [Q&A]
Is it possible to hide a plot  based on a condition?
I would like to dynamically show or hide a bottom plot (with studies) in PyQt.
This would be similar to the QT hide() and show() methods for widgets: 
https://doc.qt.io/qtforpython/PySide6/QtWidgets/QWidget.html?highlight=qwidget#PySide6.QtWidgets.PySide6.QtWidgets.QWidget.hide

I have tested the ax.hide() but it just clears the data and it still occupies the space.

Example: would like to hide the plot in the red rectangle below:
![image](https://user-images.githubusercontent.com/1431006/172460520-720a42a8-1124-4383-967c-9ba08467e5b5.png)

**Comment by highfestiva** [comment]
`hide()` unfortunately only seems to work on a last (or non-first) axis. I found a hack you might be able use, but unfortunately I don't know of a better way atm:
```python
ax2.vb.win.addItem(ax2, col=1, row=0) # auto-remove and add ax2 as first (replacing ax0)
ax3.vb.win.addItem(ax3, col=1, row=1) # auto-remove and add ax3 as second
```
It needs to take overlays into consideration, so it takes some consideration to implement inside `hide()`. But GL!
